sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorDispatcher.java

```diff
@@ -50,6 +50,7 @@
     private final ConcurrentLinkedQueue<Work> workQueue;
     private final WorkScheduler workScheduler;
     private final AtomicInteger wip = new AtomicInteger();
+    private final AtomicBoolean isIOPipeInterrupted = new AtomicBoolean();
```

Comment:
Is `AtomicBoolean` sufficient to force all threads to share the same view of memory? Is there a possibility with this design that two threads take a view of this object and both have an error and both enter the `emitError` block?

Reply:
So with the above atomic guarantee, only one thread out of many gets true for the "if" check and enters the critical section. CAS is another option, but I used getAndSet, given that's an API we use commonly in other places where atomicity is required.